app/controllers/support_interface/api_tokens_controller.rb

```diff
@@ -5,8 +5,18 @@ def index
     end
 
     def create
+      raise_error_unless_provider(params)
+
       provider = Provider.find(params[:vendor_api_token][:provider_id])
       @unhashed_token = VendorAPIToken.create_with_random_token!(provider: provider)
     end
+
+  private
+
+    def raise_error_unless_provider(params)
+      if params[:vendor_api_token][:provider_id].blank?
+        raise MissingProviderError, 'Did not select a provider'
```

Comment:
I'm still hitting this page when I test it
<img width="750" alt="image" src="https://user-images.githubusercontent.com/47917431/174564520-7331f143-e2bf-4592-aefc-61c86af9a6df.png">